SCClassLibrary/Common/Math/Rational.sc

```diff
@@ -0,0 +1,194 @@
+	
+	asInteger { ^((this.numerator / this.denominator).asInteger) }
+	
+	asInt { ^this.asInteger }
+	
+	reciprocal { ^Rational(this.denominator, this.numerator) }
+	
+	printOn { arg stream;
+		stream << "Rational( " << numerator << ", " << denominator << " )";
+	}
+	
+	hash { ^numerator.hash << 2 bitXor: denominator.hash } 
+	
+	+ { arg aNumber, adverb;
+		if(aNumber.isNumber) {
+			^this.species.new(
+				(this.numerator * aNumber.denominator) + (this.denominator * aNumber.numerator),
+				this.denominator * aNumber.denominator)
+		} {
+			^aNumber.performBinaryOpOnRational('+', this, adverb)
+		};
+	}
+	
+	- { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				(this.numerator * aNumber.denominator) - (this.denominator * aNumber.numerator),
+				this.denominator * aNumber.denominator )
+		}  {
+			^aNumber.performBinaryOpOnRational('-', this, adverb)
+		};	
+	}
+	
+	* { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				this.numerator * aNumber.numerator,
+				this.denominator * aNumber.denominator)
+		} {
+			^aNumber.performBinaryOpOnRational('*', this, adverb)
+		};
+	}
+	
+	
+	/ { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^this.species.new(
+				this.numerator * aNumber.denominator,
+				this.denominator * aNumber.numerator )
+		} {
+			^aNumber.performBinaryOpOnRational('/', this, adverb)
+		}
+	}
+	
+	== { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^((this.numerator * aNumber.denominator) == (this.denominator * aNumber.numerator))
+		} {
+			^aNumber.performBinaryOpOnRational('==', this, adverb)
+		}	
+	}
+	
+	!= { arg aNumber, adverb;
+		if(aNumber.isKindOf(Rational)) {
+			^((this.numerator * aNumber.denominator) != (this.denominator * aNumber.numerator))
+		} {
+			^aNumber.performBinaryOpOnRational('!=', this, adverb)
+		}	
+	}
+	
+	pow { arg aNumber;
+		case
+		{ aNumber == 0 } { ^(1.as(this.species)) }
+		{ aNumber > 0  } { ^(this.species.new(this.numerator.pow(aNumber.asFloat), this.denominator.pow(aNumber.asFloat))) }
+		{ aNumber < 0  } {
+			if((this.numerator == 0).not) {
+				^this.reciprocal.pow(aNumber.abs)
+			} {
+				"Numerator can't be 0".warn;
+			};
+		};
+	}
+	
+	simplify { arg maxDenominator=20;
+		^this.asFloat.asRational(maxDenominator)
+	}
+	
+	< { arg aNumber;
+		^(this.asFloat < aNumber.asFloat)
+	}
+	
+	> { arg aNumber;
+		^(this.asFloat > aNumber.asFloat)
+	}
+	
+	<= { arg aNumber;
+		^(this.asFloat <= aNumber.asFloat)
+	}
+	
+	>= { arg aNumber;
+		^(this.asFloat >= aNumber.asFloat)
+	}
+	
+	isNegative {
+		^this.numerator.isNegative
+	}
+	
+	isPositive {
+		^this.numerator.isPositive
+	}
+	
+	neg {
+		^(this * (-1))
+	}
+	
+	mod {
+		if (this.isNegative) {
+			^this.neg
+		} {
+			^this
+		};
+	}
+	
+	max { arg aNumber;
+		var rationalArray, floatArray;
+		rationalArray = [this, aNumber];
+		floatArray = rationalArray.asFloat;
+		^rationalArray.at(floatArray.indexOf(floatArray.max));
```

Comment:
wouldn't it be better if you wrote:

`^if(aNumber.asFloat > this.asFloat) { aNumber } { this }`

?

In a way, we are throwing away here the ordering features that are one of the main reasons to use rational at all. So here it might be better to implement them like `+` with full dispatch.